source/patterns/@aws-solutions-constructs/aws-fargate-s3/test/fargate-s3.test.ts

```diff
@@ -0,0 +1,406 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import '@aws-cdk/assert/jest';
+import * as defaults from '@aws-solutions-constructs/core';
+import * as cdk from "@aws-cdk/core";
+import { FargateToS3 } from "../lib";
+import * as s3 from '@aws-cdk/aws-s3';
+import * as ecs from '@aws-cdk/aws-ecs';
+
+test('New service/new bucket, public API, new VPC', () => {
+  // An environment with region is required to enable logging on an ALB
+  const stack = new cdk.Stack();
+  const publicApi = true;
+  const clusterName = "custom-cluster-name";
+  const containerName = "custom-container-name";
+  const serviceName = "custom-service-name";
+  const bucketName = "custom-bucket-name";
+  const familyName = "family-name";
+
+  const construct = new FargateToS3(stack, 'test-construct', {
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: '172.0.0.0/16' },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: familyName},
+    fargateServiceProps: { serviceName },
+    bucketProps: { bucketName },
+    logS3AccessLogs: false,
+    bucketPermissions: ['Delete', 'Put', 'Read', 'ReadWrite', 'Write']
```

Comment:
There's no check on these in any of the tests.  Including all the options will run al the code and get you 100% coverage, but without confirmation of the output it's just a gimmick to increase the coverage total.

One test should specify a subset, then confirm the result (probably have to confirm with a non-granular check like line 75.